src/Persistence.php

```diff
@@ -52,7 +52,7 @@ public static function connect($dsn, string $user = null, string $password = nul
             case 'pdo_sqlsrv':
             case 'pdo_oci':
             case 'oci8':
-                $persistence = new Persistence\Sql($dsn, $dsn['user'], $dsn['password'], $args);
+                $persistence = new Persistence\Sql($dsn, $dsn['user'] ?? null, $dsn['password'] ?? null, $args);
```

Comment:
> connection string with empty password and in such case password is not set in dsn array

why `$dsn['user'] ?? null` then?

Reply:
```http://username:@``` also don't set `password` key in dsn array and using `?? null` to avoid exception in case `password` (or user) keys are not set in dsn array.